### PR TITLE
dependencies: use poetry

### PR DESCRIPTION
**Pull request by jma**
Pipenv was used before, but often when a new version of dependency is
published, we need to adapt the Pipenv file to fix travis errors. This
is mainly due to the pipenv dependencies resolution
algorithm.

* Adapts docker files.
* Uses poetry for the travis tests.
* Uses python-dotenv to load .env and .flaskenv files and adapts the
tests accordingly.
* Updates the documentation.
* Fixes the item view and the tests detected thanks to an upgrade of
werkzeug.
* Removes bad appnope hacks by mark this macosx specific package
optional.
* Removes the setuptools manifest checks as it is not used until now and
causes problems with poetry: probably it is useless.
* Adapts all local scripts.
* Removes useless constraints.
* Adds celery configuration to load .env and .flaskenv files.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- To fix dependencies recurrents problems

## How to test?

- Install poetry: https://python-poetry.org/docs/#installation
- Remove pipenv: optional
- Install the project as usual by replacing `pipenv run` by `poetry run`.
- Test the server, and run the tests: `poetry run run-tests`.

## Available commands

* `poetry run bootstrap`
* `poetry run console`
* `poetry run run-tests`
* `poetry run tests`
* `poetry run server`
* `poetry run setup`
* `poetry run update`
* `poetry run python ./setup.py extract_messages`
* `poetry run python ./setup.py update_catalog`
* `poetry run python ./setup.py compile_catalog`

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -37,8 +37,7 @@ Requirements
 
 - ``git``
 - ``docker``, ``docker-compose``
-- ``python``, ``pip``, ``pyenv``
-- ``pipenv``
+- ``python``, ``pip``, ``poetry``
```
Comment (blankoworld) [inline INSTALL.rst:40]: Could you add pyenv to be coherent with next section (Installation) that uses pyenv?

Comment (jma) [reply INSTALL.rst:40]: Oups it is a mistake.

**Comment by blankoworld** [comment]
Server gives something like that: ```[2020-05-12 11:46:39,102] ERROR in app: Failed to initialize entry point: invenio_records = invenio_records:InvenioRecords                                                                        
Error:              
Unable to load celery application.  
The module flask_bootstrap was not found.```

**Comment by BadrAly** [comment]
The code coverage seems to go red. I do not know why.

**Pull request merged**